--- a/src/main/java/org/bukkit/craftbukkit/entity/CraftLivingEntity.java
+++ b/src/main/java/org/bukkit/craftbukkit/entity/CraftLivingEntity.java
@@ -1211,4 +_,36 @@
         return this.getHandle().canUseSlot(org.bukkit.craftbukkit.CraftEquipmentSlot.getNMS(slot));
     }
     // Paper end - Expose canUseSlot
+
+    // Noxesium start
+    @Override
+    public boolean isUsingClientsideTrident() {
+        return getHandle().isUsingClientsideTrident();
+    }
+
+    @Override
+    public void setUseClientsideTrident(boolean useClientTrident) {
+        getHandle().setUseClientsideTrident(useClientTrident);
+    }
+    // Noxesium end
 }
