--- a/src/main/java/org/bukkit/entity/LivingEntity.java
+++ b/src/main/java/org/bukkit/entity/LivingEntity.java
@@ -1468,4 +_,59 @@
      */
     boolean canUseEquipmentSlot(org.bukkit.inventory.@NotNull EquipmentSlot slot);
     // Paper end - Expose canUseSlot
+
+    // Noxesium start
+    /**
+     * Returns whether this entity is using a client-side implementation of the
+     * riptide trident mechanics, provided by Noxesium.
+     *
+     * It is assumed the necessary server rule is set in Noxesium separately.
+     * Issues are expected if this is true without Noxesium being set up properly.
+     *
+     * @return whether this entity uses the client-side trident
+     */
+    boolean isUsingClientsideTrident();
+
+    /**
+     * Sets whether this entity is using a client-side implementation of the
+     * riptide trident mechanics, provided by Noxesium.
+     *
+     * @param useClientTrident whether this entity uses the client-side trident
+     */
+    void setUseClientsideTrident(boolean useClientTrident);
+    // Noxesium end
 }
