--- a/net/minecraft/world/item/TridentItem.java
+++ b/net/minecraft/world/item/TridentItem.java
@@ -119,7 +_,7 @@
                         }
                     }
 
-                    if (tridentSpinAttackStrength > 0.0F) {
+                    if (tridentSpinAttackStrength > 0.0F && !player.isUsingClientsideTrident()) { // Noxesium - disable if using client-side trident
                         float yRot = player.getYRot();
                         float xRot = player.getXRot();
                         float f = -Mth.sin(yRot * (float) (Math.PI / 180.0)) * Mth.cos(xRot * (float) (Math.PI / 180.0));
