--- a/net/minecraft/world/entity/LivingEntity.java
+++ b/net/minecraft/world/entity/LivingEntity.java
@@ -3596,7 +_,7 @@
     protected void checkAutoSpinAttack(AABB boundingBoxBeforeSpin, AABB boundingBoxAfterSpin) {
         AABB aabb = boundingBoxBeforeSpin.minmax(boundingBoxAfterSpin);
         List<Entity> entities = this.level().getEntities(this, aabb);
-        if (!entities.isEmpty()) {
+        if (!entities.isEmpty() && !isUsingClientsideTrident()) { // Noxesium - disable spin attacks if using client tridents
             for (Entity entity : entities) {
                 if (entity instanceof LivingEntity) {
                     this.doAutoAttackOnTouch((LivingEntity)entity);
@@ -4213,9 +_,37 @@
         }
     }
 
+    // Noxesium start
+    private boolean useClientTrident = false;
+
+    public void setUseClientsideTrident(boolean useClientTrident) {
+        this.useClientTrident = useClientTrident;
+    }
+
+    public boolean isUsingClientsideTrident() {
+        return useClientTrident;
+    }
+    // Noxesium end
 
     public boolean attackable() {
         return true;